src/psyclone/psyir/transformations/increase_rank_loop_arrays_trans.py

```diff
@@ -129,69 +130,101 @@ def validate(self, node, options=None):
         # string that resolves to it
-        array_list = options.get("arrays", [])
-        if len(array_list) == 0:
+        if not isinstance(arrays, list) or len(arrays) == 0:
             raise TransformationError(
                 f"{self.name} has a mandatory 'arrays' option that "
                 f"needs to be provided to inform what arrays needs their "
                 f"rank increased.")
-        for array in array_list:
+        for array in arrays:
             if isinstance(array, str):
-                array = node.scope.symbol_table.lookup(array, otherwise=None)
-            if array is None:
-                continue
-
-            if not array.is_automatic or not array.is_array:
+                try:
+                    array = node.scope.symbol_table.lookup(array)
+                except KeyError as err:
+                    raise TransformationError(
+                        f"{self.name} provided array '{array}' does not exist"
+                        f"in this scope."
+                    ) from err
+
+            if (not isinstance(array, Symbol) or not array.is_automatic
+                    or not array.is_array):
                 raise TransformationError(
-                    f"{self.name} provided 'arrays' must be a local array "
+                    f"{self.name} provided 'arrays' must be local array "
                     f"symbols, but '{array}' is not")
 
             if array.name.lower() in codeblock_symbols:
                 raise TransformationError(
                     f"{self.name} does not support arrays that are referenced "
                     f"inside a Codeblock, but '{array.name}' is inside one.")
+            # TODO: check that the bound expressions are valid as static
+            # expressions in the declarations.
 
-            # TODO: What should happen with references outside the loop?
-
-    def apply(self, node, options=None):
+    def apply(
+        self,
+        node: Loop,
+        arrays: Optional[list[Union[Symbol, str]]] = None,
+        **kwargs
+    ):
         '''Applies the transformation.
 
-        :param node: target PSyIR node.
-        :type node: :py:class:`psyclone.psyir.nodes.Loop`
-        :param options: a dictionary with options for transformations.
-        :type options: Optional[Dict[str, Any]]
+        :param node: target Loop node.
+        :param arrays: list of arrays that will have the rank increased.
 
         '''
-        self.validate(node, options)
-
-        array_list = options.get("arrays", [])
+        self.validate(node, arrays, **kwargs)
 
-        for array in array_list:
+        for array in arrays:
             if isinstance(array, str):
                 array = node.scope.symbol_table.lookup(array, otherwise=None)
-            if array is None:
-                continue
 
-            # TODO: check that the bound expressions are valid as static
-            # expressions in the declarations.
             array.shape.append(
```

Comment:
Comment pls - "Add an additional dimension to this array with the same bounds as the target Loop"